test/pblog_test.cc

```diff
@@ -56,32 +56,34 @@ class PblogFileTest : public ::testing::Test {
     file_regions[1].size = size1;
     file_regions[1].used_size = 0;
 
-    flash_ri_ = reinterpret_cast<struct record_intf *>(
+    flash_ri_ = static_cast<struct record_intf *>(
         malloc(sizeof(struct record_intf)));
-    pblog_file_ops.priv = (void *)filename_.c_str();
-    record_intf_init(flash_ri_, file_regions, 2, &pblog_file_ops);
+    pblog_file_ops.priv = static_cast<void *>(&filename_[0]);
```

Comment:
No it has no guarantees but libstdc++ has it this way :) I figured out a way for the linter to not be annoying about the const cast.